tk1/src/pay/PaymentTransaction.java

```diff
@@ -0,0 +1,54 @@
+package pay;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.Set;
+
+import notification.MarketObserver;
+import notification.MarketSubject;
+
+public class PaymentTransaction implements MarketSubject{
+	private static int counter=0;
+	private double total;	
+	private ArrayList<MarketObserver> marketObservers;

```

Comment:
Instead of defining marketObservers as ArrayList (the concrete implementation) use List.
`List<MarketObserver> marketObservers;`
Then you can put there whatever implementation of list you want.
`marketObservers = new ArrayList<MarketObservers>();`